src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -827,20 +871,24 @@ private function getPropertyName(string $methodName, array $reflectionProperties
      * Searches for add and remove methods.
      *
      * @param \ReflectionClass $reflClass The reflection class for the given object
-     * @param array            $singulars The singular form of the property name or null
      *
      * @return array An array containing the adder and remover when found and errors
      */
-    private function findAdderAndRemover(\ReflectionClass $reflClass, array $singulars): array
+    private function findAdderAndRemover(string $property, string $camelProp, \ReflectionClass $reflClass): array
     {
+        $propertyMetadata = $this->getClassMetadata($reflClass->getName())?->getPropertyMetadataFor($property);
+        if ((null !== $addMethod = $propertyMetadata?->adder) && (null !== $removeMethod = $propertyMetadata?->remover)) {
```

Comment:
for readability (to me at least), there are many cases where checks could be simplified like this:
```suggestion
        if (($addMethod = $propertyMetadata?->adder) && ($removeMethod = $propertyMetadata?->remover)) {
```

Reply:
Done